sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringExpressions.scala

```diff
@@ -1573,8 +1573,10 @@ case class StringLPad(str: Expression, len: Expression, pad: Expression)
   override def second: Expression = len
   override def third: Expression = pad
 
-  override def dataType: DataType = str.dataType
-  override def inputTypes: Seq[AbstractDataType] = Seq(StringType, IntegerType, StringType)
+  override def dataType: DataType = CollationTypeCasts.getOutputCollation(Seq(str, pad))
```

Comment:
this is not correct

dataType should be str.dataType, and collationTypeCast should ensure that str and pad have the same dataType

Reply:
I understand now. Thanks.

Reply:
@uros-db i am not getting an exception by default. Seems collationTypeCast is not trying to cast these to the same type. Please advise.

I could override checkInputDataTypes as an alternative.

Reply:
by the way, `checkInputDataTypes` override could work theoretically, but that's absolutely not the approach we should take here

Reply:
@uros-db Thank you!  Please re-review.